### ISSUE TITLE
% unit is somehow causing "Maximum update depth exceeded"

### ISSUE THREAD

**Issue by literat** [DominicTobias/react-image-crop#364]
Hi,

we are using react-image-crop:8.6.2 in our project and if we set `unit` in crop object to "%" it ends up with "Maximum update depth exceeded".

```javascript
<ReactCrop
  src={image}
  crop={crop}
  ruleOfThirds
  onImageLoaded={onImageLoaded}
  onComplete={onCropComplete}
  onChange={onCropChange} // just setting state
  keepSelection
/>
```

```javascript
const [currentCrop, setCurrentCrop] = useState({
    unit: '%', // `'px' here just works fine`
    width: 100,
    height: 50,
    x: 25,
    y: 25,
    ...imageCrop,
  });
```

![Screenshot 2020-06-05 at 14 51 28](https://user-images.githubusercontent.com/2481010/83878693-d4e6e700-a73c-11ea-903b-46af1e762ced.png)

The last working version was 8.5.0.

Thank you for your help.

**Comment by DominicTobias** [comment]
Hi when does it break and are you doing anything non-standard in onChange or onImageLoad?

**Comment by literat** [comment]
I think that nothing bad. Just state and some Math.

```javascript
const handleCropChange = (crop, percentCrop) => {
    setCurrentCrop(percentCrop);
};
```

```javascript
const handleCropComplete = (crop, percentCrop) => {
    setCurrentCrop(percentCrop);
    setCropRealPosition({
      x1: Math.ceil((imageDimensions.width * percentCrop.x) / 100),
      y1: Math.ceil((imageDimensions.height * percentCrop.y) / 100),
      x2: Math.ceil(
        (imageDimensions.width * (percentCrop.width + percentCrop.x)) / 100,
      ),
      y2: Math.ceil(
        (imageDimensions.height * (percentCrop.height + percentCrop.y)) / 100,
      ),
    });
  };
```

We are also setting custom styles for img tag:

```less
.image-editor {
  img {
    width: 560px;
    margin-left: 0px;
    transform: none;
  }
}

#cropper-wrapper {
  img {
    max-width: 100%;
  }
}
```

**Comment by DominicTobias** [comment]
Please remove the custom styles except for `max-width: 100%;` and try again. I will be making a fix for this but looking for a workaround.

**Comment by DominicTobias** [comment]
Please try 8.6.3

**Comment by literat** [comment]
Great! Removing the style fix the problem. Thank you.

Component breaks after +/- 1s. I see the image loaded for a while.

![Screenshot 2020-06-05 at 15 39 17](https://user-images.githubusercontent.com/2481010/83882941-1aa6ae00-a743-11ea-8ce5-08c2203a685a.png)

**Comment by DominicTobias** [comment]
Ok cool, this should be fixed in 8.6.3 anyway. If you do want to limit the crop width though I recommend just doing it on a container element of the crop. Closing but please feel free to add any issues with it.

**Comment by literat** [comment]
Thank you. 8.6.3 is working just fine. 👍
